app/controllers/api/attempts/events_controller.rb

```diff
@@ -19,6 +22,35 @@ def status
           reason: :not_yet_implemented,
         }
       end
+
+      private
+
+      def authenticate_client
+        bearer, issuer, token = request.authorization&.split(' ', 3)
+        if bearer != 'Bearer' ||
+           config_data(issuer).blank? || !valid_auth_token?(token, issuer)
+
+          render json: { status: 401, description: 'Unauthorized' }, status: :unauthorized
+        end
+      end
+
+      def service_provider(issuer)
+        !IdentityConfig.store.allowed_attempts_providers.map(&:issuer).include?(issuer)
+      end
+
+      def valid_auth_token?(token, issuer)
+        config_data(issuer)[:token] == token
```

Comment:
We should consider using `ActiveSupport::SecurityUtils.secure_compare`. Even better would be to store a hashed version (which we did a half-hearted version of [here](https://github.com/18F/identity-idp/pull/7888))

Reply:
okay what do you uh, think of https://github.com/18F/identity-idp/pull/11889/commits/5ec84ca1677dd15d6cbf6ac0ef3011b48f556131? the `IdentityConfig` stuff feels pretty bad but i think necessary to accommodate multiple partners and multiple tokens

Reply:
update, we are going to assume a prehashed token, and keep track of the salt within the config.